From 132df9829fa4c697e0b9701871888708973f9123 Mon Sep 17 00:00:00 2001
From: Gerry Kopec <Gerry.Kopec@windriver.com>
Date: Fri, 16 Aug 2019 14:29:46 -0400
Subject: [PATCH] Enable override of rabbitmq probe parameters

Add variables for initial delay, period and timeout for rabbitmq
liveness and readiness probes.

Signed-off-by: Gerry Kopec <Gerry.Kopec@windriver.com>
[ fix duplicate yaml keys ]
Signed-off-by: Thales Elero Cervi <thaleselero.cervi@windriver.com>
Change-Id: I3db83ffd3c7856b099cc6fb488e3df4e8da7fb2c
---
 rabbitmq/templates/statefulset.yaml | 10 ++++++----
 rabbitmq/values.yaml                |  8 ++++++++
 2 files changed, 14 insertions(+), 4 deletions(-)

diff --git a/rabbitmq/templates/statefulset.yaml b/rabbitmq/templates/statefulset.yaml
index 35a482c1..5d013a1e 100644
--- a/rabbitmq/templates/statefulset.yaml
+++ b/rabbitmq/templates/statefulset.yaml
@@ -243,14 +243,16 @@ spec:
               value: {{ $envAll.Values.io_thread_pool.size | quote }}
 {{- end }}
           readinessProbe:
-            initialDelaySeconds: 10
-            timeoutSeconds: 10
+            initialDelaySeconds: {{ $envAll.Values.pod.probes.readiness.initialDelaySeconds }}
+            periodSeconds: {{ $envAll.Values.pod.probes.readiness.periodSeconds }}
+            timeoutSeconds: {{ $envAll.Values.pod.probes.readiness.timeoutSeconds }}
             exec:
               command:
               - /tmp/rabbitmq-readiness.sh
           livenessProbe:
-            initialDelaySeconds: 60
-            timeoutSeconds: 10
+            initialDelaySeconds: {{ $envAll.Values.pod.probes.liveness.initialDelaySeconds }}
+            periodSeconds: {{ $envAll.Values.pod.probes.liveness.periodSeconds }}
+            timeoutSeconds: {{ $envAll.Values.pod.probes.liveness.timeoutSeconds }}
             exec:
               command:
               - /tmp/rabbitmq-liveness.sh
diff --git a/rabbitmq/values.yaml b/rabbitmq/values.yaml
index 6e6978c0..e24db892 100644
--- a/rabbitmq/values.yaml
+++ b/rabbitmq/values.yaml
@@ -54,6 +54,14 @@ forceBoot:
 
 pod:
   probes:
+    readiness:
+      initialDelaySeconds: 10
+      periodSeconds: 10
+      timeoutSeconds: 10
+    liveness:
+      initialDelaySeconds: 30
+      periodSeconds: 10
+      timeoutSeconds: 10
     prometheus_rabbitmq_exporter:
       rabbitmq_exporter:
         readiness:
-- 
2.25.1

